coderd/workspaces.go

```diff
@@ -839,6 +873,27 @@ func requestTemplate(ctx context.Context, rw http.ResponseWriter, req codersdk.C
 	return template, true
 }
 
+func claimPrebuild(ctx context.Context, claimer prebuilds.Claimer, db database.Store, logger slog.Logger, req codersdk.CreateWorkspaceRequest, owner workspaceOwner) (*database.Workspace, error) {
+	prebuildsCtx := dbauthz.AsPrebuildsOrchestrator(ctx)
+
+	// TODO: do we need a timeout here?
+	claimCtx, cancel := context.WithTimeout(prebuildsCtx, time.Second*10)
+	defer cancel()
+
+	claimedID, err := claimer.Claim(claimCtx, db, owner.ID, req.Name, req.TemplateVersionPresetID)
```

Comment:
`ClaimPrebuiltWorkspace` simply executes an `UPDATE` which should not require a separate timeout.

Reply:
done, now I'm just using `prebuildsCtx` instead of `claimCtx`